packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -365,7 +367,15 @@ export class DocumentDeltaConnection
                         description.target = undefined;
                     }
                 } catch(_e) {}
-                fail(true, this.createErrorObject("connectError", error));
+                if (!this.socket.io.reconnection() ||
+                    this.reconnectAttempts >= (this.socket.io.reconnectionAttempts() ?? 0)) {
```

Comment:
It would be great from start to incorporate proper telemetry into this flow:
- events for reconnection attempts (and their duration / protocol)
- Once connection is established - to log (in some form) of long polling or websocket is used.
Ideally the form telemetry takes is left to the caller, so maybe we just need hooks for each driver to be able to leverage them? Maybe not, and this layer just issuing events is fine.
@jatgarg , @markfields - FYI

Reply:
Seems like since the logic is shared the logging can live here too, but maybe I'm not seeing the benefits of each driver having more flexibility.

I'll be curious to see how often this happens, and for tenants configured this way how spammy these logs are.  If there's some other error that triggers this reconnect but never succeeds, then we'll just sit and log the same error every 8 seconds (see `connectCore`).  #8411 would help...

Reply:
Created #8968 to track this as a follow up for this PR